bin/codemods/run.js

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env node
+
+/**
+ * External dependencies
+ */
+const fs = require( 'fs' );
+const path = require( 'path' );
+const child_process = require( 'child_process' );
+const glob = require( 'glob' );
+
+/**
+ * Internal dependencies
+ */
+const config = require( './src/config' );
+
+function getLocalCodemodFileNames() {
+	// Returns all JS files in bin/codemods/src folder, except for config and helpers.
+	return glob.sync( 'bin/codemods/src/*.js' )
+		.filter( name => name !== 'config.js' ) // exclude utility files
+		.map( name => name.replace( /\.js$/, '' ) ); // strip extension from filename
+}
+
+function getValidCodemodNames() {
+	return [
+		...getLocalCodemodFileNames(),
+		...Object.getOwnPropertyNames( config.codemodArgs ),
+	].map( name => '- ' + name ).sort();
+}
+
+function generateBinArgs( name ) {
+	if ( config.codemodArgs.hasOwnProperty( name ) ) {
+		// Is the codemod defined in the codemodArgs object?
+		return config.codemodArgs[ name ];
+	}
+
+	if ( getLocalCodemodFileNames().includes( name ) ) {
+		// Is the codemod a local script defined in bin/codemods/src folder?
+		return [
+			`--transform=bin/codemods/src/${ name }.js`,
+		];
+	}
+
+	throw new Error(
+		`"${ name }" is an unrecognized codemod.`
+	);
+}
+
+function runCodemod() {
+	const args = process.argv.slice( 2 );
```

Comment:
another fun way I like to write out these kinds of situations is with destructuring to provide additional context…

```js
const [ /* binary name */, /* whatever this arg is */, ...args ] = process.argv;
```